### PR TITLE
MAINT: work around lstsq driver selection issue

### PR DESCRIPTION
**Pull request by ev-br**
Switch the default driver to gelss if gelsd is buggy. As suggested in https://github.com/scipy/scipy/issues/5539#issuecomment-159925122. Closes gh-5539.

### CODE REVIEW

File: scipy/linalg/basic.py [hunk 1]
```diff
@@ -826,11 +830,14 @@ def lstsq(a, b, cond=None, overwrite_a=False, overwrite_b=False,
     if m != b1.shape[0]:
         raise ValueError('incompatible dimensions')
 
-    if lapack_driver not in ('gelsd', 'gelsy', 'gelss'):
-        raise ValueError('LAPACK driver "%s"is not found' % lapack_driver)
+    driver = lapack_driver
```
Comment (ewmoore) [inline scipy/linalg/basic.py]: `lapack_driver` could be changed to just `driver` above too.  This is all unreleased.

Comment (ewmoore) [inline scipy/linalg/basic.py]: Nevermind.  I see what you're doing below.

**Comment by ewmoore** [comment]
LGTM.

File: scipy/linalg/tests/test_basic.py [hunk 2]
```diff
@@ -738,16 +739,20 @@ def test_simple_exact(self):
...
                             b1 = b.copy()
-                            out = lstsq(a1, b1, lapack_driver=lapack_driver,
-                                        overwrite_a=overwrite,
-                                        overwrite_b=overwrite)
+                            try:
+                                out = lstsq(a1, b1, lapack_driver=lapack_driver,
+                                            overwrite_a=overwrite,
+                                            overwrite_b=overwrite)
+                            except LstsqLapackError:
+                                continue
```
Comment (pv) [inline scipy/linalg/tests/test_basic.py]: `if lapack_driver is None: raise`?

Comment (ev-br) [inline scipy/linalg/tests/test_basic.py]: "Constant vigilance!" :-)
 Done.

**Comment by rgommers** [comment]
This improves things a lot, but I'm getting 6 test failures in ``TestLstsq``. Mainly for float32 tests it looks like. Do you see those as well @ev-br?

**Comment by ev-br** [comment]
Grrrr... Should be fixed now, also on numpy-vendor binaries.

**Comment by rgommers** [comment]
LGTM now, merged. Thanks Evgeni!

**Pull request merged**